src/glados/audio_io/sounddevice_io.py

```
@@ -0,0 +1,236 @@
+    def stop_listening(self) -> None:
+        """Stop capturing audio and clean up resources.
+
+        Stops the input stream if it's active and releases associated resources.
+        This method should be called when audio input is no longer needed or
+        before application shutdown.
+        """
+        if self.input_stream is not None:
+            try:
+                self.input_stream.stop()
+                self.input_stream.close()
+            except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)